src/shared/openssl-util.h

```diff
@@ -193,12 +181,13 @@ struct OpenSSLAskPasswordUI {
         UI_METHOD *method;
 };
 
-int openssl_ask_password_ui_new(OpenSSLAskPasswordUI **ret);
+int openssl_ask_password_ui_new(const AskPasswordRequest *request, OpenSSLAskPasswordUI **ret);
 
 static inline OpenSSLAskPasswordUI* openssl_ask_password_ui_free(OpenSSLAskPasswordUI *ui) {
         if (!ui)
                 return NULL;
 
+        UI_set_default_method(UI_OpenSSL());
```

Comment:
maybe validate that UI_get_default_method() still points to our implementation with an assert()?